tests/assert/nested_array_equals.sql

```diff
@@ -0,0 +1,23 @@
+/*
+This assumes a convention where a nested array is actually a STRUCT with an `arr` field.
```

Comment:
In generating BQ schemas from JSONSchema, we have a convention of transforming nested arrays where we create a struct with a field called `list` rather `arr`. I would like to stick with that convention and assume the field name is `list`.

See, for example, https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/PubsubMessageToObjectNode.java#L312